### PR TITLE
webidl: add support for static methods and attributes

### LINKED ISSUE #251: Support emitting getters for static WebIDL attributes
https://heycam.github.io/webidl/#idl-attributes

Given

```webidl
interface Foo {
  static attribute long bar;
};
```

we should emit a `fn get_bar() -> i32` method to get `Foo`'s static `bar` attribute.

This is similar to #250 except that there is no `&self` parameter.

### PR DESCRIPTION
**Pull request by ohanar**
Fixes #251.

### CODE REVIEW

File: tests/all/webidl/simple.rs [hunk 1]
```diff
@@ -176,3 +176,59 @@ fn named_constructor() {
...
+
+            use wasm_bindgen::prelude::*;
+
+            pub mod foo;
+
+            use foo::Foo;
+
+            #[wasm_bindgen]
+            pub fn test() {
+                let tmp = Foo::swap(3.14159) == 0.;
```
Comment (fitzgen) [inline tests/all/webidl/simple.rs]: Nitpick: we should use `assert_eq!(..)` here

File: tests/all/webidl/simple.rs [hunk 2]
```diff
@@ -176,3 +176,59 @@ fn named_constructor() {
...
+                assert!(tmp);
+                let tmp = Foo::swap(2.71828) == 3.14159;
+                assert!(tmp);
+                let tmp = Foo::swap(2.71828) != 3.14159;
```
Comment (fitzgen) [inline tests/all/webidl/simple.rs]: And `assert_ne!(..)` here

**Comment by fitzgen** [comment]
@ohanar, would you be interested in joining as a collaborator for wasm-bindgen? Specifically with regards to the webidl frontend, since you've now written most of the code that makes it up as well as its testing infrastructure :)

Responsibilities involved are basically what's listed [here](https://github.com/rustwasm/team/blob/master/GOVERNANCE.md#repositories), and specifically reviewing pull requests related to webidl from time to time.

**Comment by ohanar** [comment]
Nitpick addressed. There is a weird issue (at least on my machine) where, in the testing framework, asserts don't ever fail without binding to a temporary variable. I had assumed this extended to `assert_eq` and `assert_ne`, but it doesn't seem to.

I would be happy to be a collaborator. As you said it sounds like my main responsibility would be in reviewing pull requests, which I am happy to do.

**Comment by fitzgen** [comment]
Awesome :) Invite sent!

**Pull request merged**